pennylane/operation.py

```diff
@@ -1025,6 +1029,29 @@ def generator(self):  # pylint: disable=no-self-use
         """
         raise GeneratorUndefinedError(f"Operation {self.name} does not have a generator")
 
+    def pow(self, n):
+        """A new operator equal to this one raised to the given power.
+
+        This *developer* method should be used to define shortcuts and easier decompositions.
+        For example, this method may specify how a Pauli gate raised to an even power is equal
+        to nothing, or we can exponentiate a rotation gate by multiplying the angle by the exponent.
+        This method should *not* return default behavior like a list of the same gate many times.
+
+        Args:
+            n (float): exponent for the operator
+
+        Returns:
+            list[:class:`~.operation.Operator`]
+
+        """
+        # Child methods may call super().pow(n%period)
+        # Hence we define 0 and 1 special cases here.
+        if n == 0:
+            return []
+        if n == 1:
+            return [self.__copy__()]
+        raise PowUndefinedError
+
```

Comment:
About codecov complaining: We usually don't test `NotImplementedError`s, but I think these special errors are important to test, because other code relies on it with try-error statements